addons/website_form/static/src/snippets/s_website_form/options.js

```diff
@@ -1025,13 +1046,22 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
         if (activeField.type !== field.type) {
             field.value = '';
         }
-        const htmlField = this._renderField(field);
-        [...this.$target[0].childNodes].forEach(node => node.remove());
-        [...htmlField.childNodes].forEach(node => this.$target[0].appendChild(node));
-        [...htmlField.attributes].forEach(el => this.$target[0].removeAttribute(el.nodeName));
-        [...htmlField.attributes].forEach(el => this.$target[0].setAttribute(el.nodeName, el.nodeValue));
+        const fieldEl = this._renderField(field);
+        this._replaceFieldElement(fieldEl);
         this.$target[0].querySelectorAll('input.datetimepicker-input').forEach(el => el.value = field.propertyValue);
```

Comment:
Can't see your diff because you rebased at the same time as resolving the conflict but I guess this was part of the conflict? :thinking: Any reason this should not be in `replaceFieldElement`?

Reply:
What if you juste move `const fieldEl = this._renderField(field);` inside that function too? Then you pass `field` and not `fieldEl` to the method and you can then also move `this.$target[0].querySelectorAll('input.datetimepicker-input').forEach(el => el.value = field.propertyValue);`

Something like

```javascript

onClone() {
    const field = this._getActiveField();
    this._replaceFieldElement(field);
},
replaceField: async function (field) {
    await this._fetchFieldRecords(field);
    const activeField = this._getActiveField();
    if (activeField.type !== field.type) {
        field.value = '';
    }
    this._replaceFieldElement(field);
}
_replaceFieldElement(field) {
    const fieldEl = this._renderField(field);
    [...this.$target[0].childNodes].forEach(node => node.remove());
    [...fieldEl.childNodes].forEach(node => this.$target[0].appendChild(node));
    [...fieldEl.attributes].forEach(el => this.$target[0].removeAttribute(el.nodeName));
    [...fieldEl.attributes].forEach(el => this.$target[0].setAttribute(el.nodeName, el.nodeValue));
    this.$target[0].querySelectorAll('input.datetimepicker-input').forEach(el => el.value = field.propertyValue);
},
```

Reply:
I was thinking about moving it inside the _renderField method, like that :
```
    /**
     * @private
     * @param {Object} field
     * @returns {HTMLElement}
     */
    _renderField: function (field) {
        if (!field.id) {
            field.id = this._generateUniqueID();
        }
        const template = document.createElement('template');
        template.innerHTML = qweb.render("website_form.field_" + field.type, {field: field}).trim();
        if (field.description && field.description !== true) {
            $(template.content.querySelector('.s_website_form_field_description')).replaceWith(field.description);
        }
        template.content.querySelectorAll('input.datetimepicker-input').forEach(el => el.value = field.propertyValue);
        return template.content.firstElementChild;
    },
    ```
 I didn't push yet because a traceback is thrown in saas-14.4 when cloning a form field in some cases. In other cases it just doesn't duplicate the field. It seems to be related to call to `this.options.wysiwyg.odooEditor.historyStep();` made in `SnippetEditor.clone()`. I am still investigating so that the fix would be part of this PR